app/controllers/hyku/invitations_controller.rb

```diff
@@ -26,12 +27,15 @@ def create
 
       yield resource if block_given?
 
-      # Override destination as this was a success either way
-      if is_flashing_format? && resource.invitation_sent_at
-        set_flash_message :notice, :send_instructions, email: resource.email
-      end
-      if method(:after_invite_path_for).arity == 1
-        respond_with resource, location: after_invite_path_for(current_inviter)
+      if resource_invited
+        if is_flashing_format? && self.resource.invitation_sent_at
+          set_flash_message :notice, :send_instructions, email: self.resource.email
+        end
+        if self.method(:after_invite_path_for).arity == 1
+          respond_with resource, location: after_invite_path_for(current_inviter)
+        else
+          respond_with resource, location: after_invite_path_for(current_inviter, resource)
+        end
       else
         respond_with resource, location: after_invite_path_for(current_inviter, resource)
       end
```

Comment:
```suggestion
      if resource_invited && is_flashing_format? && self.resource.invitation_sent_at
        set_flash_message :notice, :send_instructions, email: self.resource.email
      elsif resource_invited && self.method(:after_invite_path_for).arity == 1
        respond_with resource, location: after_invite_path_for(current_inviter)
      else
        respond_with resource, location: after_invite_path_for(current_inviter, resource)
      end
```

Reply:
Thank you, `if resource_invited` is a wrapper and cannot be combined with the rest of the second if statement.